packages/block-library/src/image/image.js

```diff
@@ -365,6 +369,18 @@ export default function Image( {
 		availableUnits: [ 'px' ],
 	} );
 
+	const lightboxSetting = useSetting( 'lightbox' );
+
+	const lightboxChecked = useMemo( () => {
+		// If the lightbox is set in the block attributes, use that.
+		if ( lightbox?.enabled ) return true;
+
+		// If the lightbox is NOT set in the block attributes AND it IS enabled in
+		// the settings, use that.
+		if ( ! lightbox && lightboxSetting?.enabled ) return true;
```

Comment:
What about someone that wants to explicitly disable the lightbox even if set to true globally? 

I'm thinking we should differentiate between "undefined" and "false" at the block level. 

 - undefined would mean inherit global value
 - false would mean explicitly disabled regardless of the global value

Reply:
> What about someone that wants to explicitly disable the lightbox even if set to true globally?

In my testing, that's how the current syntax works. The block inherits from the global value, and one is also able to override it with the local block setting.

Is the syntax unclear?

Reply:
But by reading the code, this use-case is not possible:

 - I want to explicitly disable the lightbox for a particular image (so block attribute set to false), It doesn't matter if theme said that "lightbox" is enabled globally or not, I want this particular block instance to not have lightbox.

I don't think I can do that at the moment.

Reply:
Hmm, I believe this use is covered. Here I've recorded a video where I set the block attribute to false while the global setting is undefined. Is this what you mean?

https://github.com/WordPress/gutenberg/assets/5360536/1d8942ab-dda5-4f37-a119-17b4ac80af8c

Reply:
Hmm, I believe this use is covered. Here I've recorded a video where I set the block attribute to false while the global setting is undefined. Is this what you mean?

No I mean when I set the block attribute to "false" while the global setting is "true". I believe it's the block setting that should win in this case.

Reply:
In other words, I believe the global setting should act as a "default" value for the block attribute but that any explicit value for the block attribute (true or false) should always be used regardless of what the theme.json says.

Reply:
> In other words, I believe the global setting should act as a "default" value for the block attribute but that any explicit value for the block attribute (true or false) should always be used regardless of what the theme.json says.

It does work exactly like this.

> I'm thinking we should differentiate between "undefined" and "false" at the block level.
> 
> undefined would mean inherit global value
> false would mean explicitly disabled regardless of the global value

I think I see what causes the confusion. You're suggesting we use `lightbox.enabled = undefined` as a third "unset" state. We're already doing this, but [setting](https://github.com/WordPress/gutenberg/blob/03e60d715d877554d4e82ebc727833c07e730b2d/packages/block-library/src/image/image.js#L543-L544) the whole lightbox attribute as `undefined`. In other words:

- `lightbox.enabled = false` - disable the lightbox for the block regardless of the global styles/theme.json
- `lightbox.enabled = true` - enable the lightbox for the block regardless of the global styles/theme.json
- `lightbox = undefined` - inherit the value from Global Styles

Reply:
You're right, it does work but I believe there a few issues with how it's implemented:

 - The obvious one is that, there's no way in the UI for the user to know that the property is "explicitly disabled" or "inheriting the theme setting". In both cases the setting is empty. For instance, imagine you enable the setting and then decide to disable it again => what's the expectation here? Are you back to "inherit" or are you saying "disable" . I believe these two states are different and should be clarified to the user in the UI.

 - The second problem is more subtle and is related to the code. Basically in the client, when parsing blocks we have "default values" that are taken into consideration and added to the block object which means that if your default value for "enabled" is "false", there's no way for you to know if it's "explicitly disabled" or "using default value". In the server though, and since we don't parse the blocks fully, the attribute is "undefined" unless you explicitly disable it. Meaning in the server you can make the difference between the two (which is why it seems to work as intended right now)
 
 So I see two actions items personally here:

 - Clarify in the UI the difference between "disabled" and "inherit" (May need some design work).
 - In the attribute definition, drop the default value (in other words, the default value should be "inherit" in both client and server)

----

The reason I'm insisting here is because this setting is going to create a precedent both in terms of UI and behavior, it's the first "block attribute" that can be enabled globally in theme.json and modified locally as well. (I think layout is similar but its UI is probably not the best).

Also, in global styles, we already have several issues that were raised because of this: in global styles, in most UI, we have "reset" links to reset to the theme.json provided values but we don't have the UI to explicitely reset styles to "undefined". (We need to solve that at some point and your work here is going to help us I believe).

Reply:
> The obvious one is that, there's no way in the UI for the user to know that the property is "explicitly disabled" or "inheriting the theme setting". In both cases the setting is empty. For instance, imagine you enable the setting and then decide to disable it again => what's the expectation here? Are you back to "inherit" or are you saying "disable" . I believe these two states are different and should be clarified to the user in the UI.

Ok thanks for the feedback. I'll take a look at the code more closely soon but here I'm pinging in @jameskoster and @jasmussen for thoughts on how we should show that inheritance is happening in the UI.

Reply:
> For instance, imagine you enable the setting and then decide to disable it again => what's the expectation here? Are you back to "inherit" or are you saying "disable" . I believe these two states are different and should be clarified to the user in the UI.

You're right, the UX is not the best here. In this case the user would **disable** the lightbox.

If they want to "go back to 'inherit'", they should use the "RESET" in the ToolsPanel (the 3 dots). But this menu is not very discoverable.

> we don't have the UI to explicitely reset styles to "undefined"

What do you mean by that? Is that not the same as using the "RESET" links?

I see that Artemio has already pinged Design so let's wait for their input, but I agree that making the "inherited" state more explicit is better UX 👍

Reply:
Oh yes, just remembered the "RESET" link, I guess that's how styles works too, it's soo bad though and also for some controls it actually also removes the control from the UI. 

So yeah very confusing for users as the presence of a link means "inherit" or "not" :).

Reply:
Designs to handle this have been [proposed here](https://github.com/WordPress/gutenberg/issues/53403#issuecomment-1709780982).

As per the discussion, it looks like we'll ship with the UI as is for now and improve the experience in a future release.